File: samples/CSharp/AdventureWorksLT/SalesOrderInfo.cs
using DevZest.Data.Annotations;

namespace DevZest.Samples.AdventureWorksLT
{
    [InvisibleToDbDesigner]
    public class SalesOrderInfo : SalesOrderBase<SalesOrderInfoDetail>
    {
        static SalesOrderInfo()
        {
            RegisterProjection((SalesOrderInfo _) => _.Customer);
            RegisterProjection((SalesOrderInfo _) => _.ShipToAddress);
            RegisterProjection((SalesOrderInfo _) => _.BillToAddress);
        }

        public Customer.Lookup Customer { get; private set; }
        public Address.Lookup ShipToAddress { get; private set; }
        public Address.Lookup BillToAddress { get; private set; }
    }
}

File: samples/AdventureWorksLT.Data.SqlServer.VisualBasic/AdventureWorksLT.Data/SalesOrderInfo.vb
Imports DevZest.Samples.AdventureWorksLT

<InvisibleToDbDesigner>
Public Class SalesOrderInfo
    Inherits SalesOrderBase(Of SalesOrderInfoDetail)

    Shared Sub New()
        RegisterProjection(Function(x As SalesOrderInfo) x.Customer)
        RegisterProjection(Function(x As SalesOrderInfo) x.ShipToAddress)
        RegisterProjection(Function(x As SalesOrderInfo) x.BillToAddress)
    End Sub

    Private m_Customer As Customer.Lookup
    Public Property Customer As Customer.Lookup
        Get
            Return m_Customer
        End Get
        Private Set
            m_Customer = Value
        End Set
    End Property

    Private m_ShipToAddress As Address.Lookup
    Public Property ShipToAddress As Address.Lookup
        Get
            Return m_ShipToAddress
        End Get
        Private Set
            m_ShipToAddress = Value
        End Set
    End Property

    Private m_BillToAddress As Address.Lookup
    Public Property BillToAddress As Address.Lookup
        Get
            Return m_BillToAddress
        End Get
        Private Set
            m_BillToAddress = Value
        End Set
    End Property
End Class
